models/public/googlenet-v4-tf/freeze.py.patch

```diff
@@ -0,0 +1,68 @@
+diff --git a/research/slim/freeze.py b/research/slim/freeze.py
+new file mode 100644
+index 0000000..411001d
+--- /dev/null
++++ b/research/slim/freeze.py
+@@ -0,0 +1,62 @@
++import argparse
++import os
++import tensorflow as tf
++from tensorflow.python.framework import graph_io
++import subprocess
++
++from nets import nets_factory
++
++def parse_args():
++    """Parse input arguments"""
++
++    parser = argparse.ArgumentParser(description='Freeze models from TF Slim')
++
++    parser.add_argument('--ckpt', type=str, required=True,
++                        help='Path to .ckpt file')
++    parser.add_argument('--name', type=str, required=True,
++                        help='Model name')
++    parser.add_argument('--num_classes', type=int, required=True,
++                        help='Number of classes')
++    parser.add_argument('--output', type=str, required=False,
++                        help='Name of output node')
++    return parser.parse_args()
++
++def convert(model_name, ckpt, num_classes, output=None, size=None):
++    tf.logging.set_verbosity(tf.logging.INFO)
++
++    with tf.Graph().as_default() as graph:
++        network_fn = nets_factory.get_network_fn(model_name, num_classes=num_classes, is_training=False)
++
++        size = size or network_fn.default_image_size
++        placeholder = tf.placeholder(name='input', dtype=tf.float32, shape=(1, size, size, 3))
++        logits, _ = network_fn(placeholder)
++
++        pb = ckpt.replace('ckpt', 'pb')
++        frozen = ckpt.replace('.ckpt', '_frozen.pb')
```

Comment:
There's a weird inconsistency here. You append `_frozen.pb` to the base name, but the file that you uploaded is called `inception_v4.frozen.pb`. You should make it so that the file is saved with the same name that you uploaded it with.